appliance/mongodb/cmd/flynn-mongodb/main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"encoding/binary"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/flynn/flynn/appliance/mongodb"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/random"
+	"github.com/flynn/flynn/pkg/shutdown"
+	sd "github.com/flynn/flynn/pkg/sirenia/discoverd"
+	"github.com/flynn/flynn/pkg/sirenia/state"
+	"gopkg.in/inconshreveable/log15.v2"
+)
+
+const (
+	mongoIdKey = "MONGODB_ID"
+)
+
+func main() {
+	serviceName := os.Getenv("FLYNN_MONGO")
+	if serviceName == "" {
+		serviceName = "mongodb"
+	}
+	singleton := os.Getenv("SINGLETON") == "true"
+	password := os.Getenv("MONGO_PWD")
+	httpPort := os.Getenv("HTTP_PORT")
+	ip := os.Getenv("EXTERNAL_IP")
+	if httpPort == "" {
+		httpPort = "27018"
+	}
+	serverId := ipToId(net.ParseIP(ip))
+
+	const dataDir = "/data"
+	idFile := filepath.Join(dataDir, "instance_id")
+	idBytes, err := ioutil.ReadFile(idFile)
+	if err != nil && !os.IsNotExist(err) {
+		shutdown.Fatalf("error reading instance ID: %s", err)
+	}
+	id := string(idBytes)
+	if len(id) == 0 {
+		id = random.UUID()
+		if err := ioutil.WriteFile(idFile, []byte(id), 0644); err != nil {
+			shutdown.Fatalf("error writing instance ID: %s", err)
+		}
+	}
+
+	keyFile := filepath.Join(dataDir, "Keyfile")
```

Comment:
The `Keyfile` thing is MongoDB convention. It may be possible to change the name with a flag but I figured it was easier to leave as is.